### PR TITLE
Increase unit test coverage

### PR DESCRIPTION
**Pull request by matt-rich**

### CODE REVIEW

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/EventFactoryTest.java [hunk 1]
```diff
@@ -39,4 +44,13 @@ public void shouldCreateCakeOrderedEvent() throws Exception {
         assertThat(cakeOrdered.getDeliveryDate(), is(ZonedDateTime.of(2016, 01, 14, 18, 15, 3, 123, ZoneId.of("UTC"))));
 
     }
+
+    @Test (expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionOnMapperIOException() throws IOException {
+        ObjectMapper mockedMapper = Mockito.mock(ObjectMapper.class);
```
Comment (mapingo) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/EventFactoryTest.java]: Could this be final

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    private EventSource eventSource;
+
+    @Mock
+    private EventStream eventStream;
+
+    @Mock
+    private AggregateService aggregateService;
+
+    @Mock
+    EventFactory eventFactory;
```
Comment (mapingo) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java]: private?

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        final String deliveryDate = ZonedDateTimes.toString(new UtcClock().now());
+
+        final JsonEnvelope command = envelope()
+                .with(metadataWithRandomUUID(COMMAND_NAME))
+                .withPayloadOf(ORDER_ID, "orderId")
+                .withPayloadOf(RECIPE_ID, "recipeId")
+                .withPayloadOf(deliveryDate, "deliveryDate")
+                .build();
+
+        CakeOrdered cakeOrdered = new CakeOrdered(ORDER_ID, RECIPE_ID, ZonedDateTimes.fromString(deliveryDate));
```
Comment (mapingo) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java]: final?

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java [hunk 4]
```diff
@@ -1,8 +1,42 @@
...
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
 public class CakeMadeEventListenerTest {
 
+    @Mock
+    CakeRepository cakeRepository;
+
+    @Mock
+    JsonObjectToObjectConverter converter;
```
Comment (mapingo) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java]: private? and above

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java [hunk 5]
```diff
@@ -1,8 +1,42 @@
...
+
+    @InjectMocks
+    private CakeMadeEventListener cakeMadeEventListener = new CakeMadeEventListener();
+
+    @Test
+    public void shouldSaveCake() {
+        JsonEnvelope envelope = Mockito.mock(JsonEnvelope.class);
+        Cake cake = Mockito.mock(Cake.class);
```
Comment (mapingo) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java]: final and above

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                streamContaining(
+                        jsonEnvelope(
+                                withMetadataEnvelopedFrom(command)
+                                        .withName(EVENT_NAME),
+                                payloadIsJson(allOf(
+                                        withJsonPath("$.orderId", equalTo(ORDER_ID.toString())),
+                                        withJsonPath("$.recipeId", equalTo(RECIPE_ID.toString())),
+                                        withJsonPath("$.deliveryDate", equalTo(deliveryDate))
+                                )))/*
+                                .thatMatchesSchema()
```
Comment (mapingo) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java]: can this be removed or reinstated?

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/EventFactoryTest.java (hunk 1 above)
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/EventFactoryTest.java]: done

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java (hunk 2 above)
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java]: done

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java (hunk 6 above)
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java]: Removed

File: example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java (hunk 3 above)
Comment (matt-rich) [inline example-context/example-service/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/OrderCakeCommandHandlerTest.java]: Done

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java (hunk 4 above)
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java]: Done

File: example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java (hunk 5 above)
Comment (matt-rich) [inline example-context/example-service/example-event/example-event-listener/src/test/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListenerTest.java]: Done

**Pull request merged**